script.js

```diff
@@ -1,4 +1,242 @@
-  var myOutputValue = 'hello world';
+  var myOutputValue = "";
+
+  // user has a choice to toggle between auto or normal game mode
+  // if game mode = normal, player gets to select dice1 or dice 2, highest wins
+  // if game mode = auto, player number is auto generated, lowest wins
+
+  if (input.toLowerCase() == "normal") {
+    gameMode = "player 1";
+    return `You are now in normal mode, click submit to continue!`;
+  }
+
+  if (input.toLowerCase() == "auto") {
+    gameMode = "auto";
+    return `You are now in auto mode, click submit to continue!`;
+  }
+  // start of normal game mode, generate roll dice for player 1
+  if (gameMode == "player 1") {
+    newDiceRolls = getDiceRoll();
+    gameMode = "wait for player 1 choice";
+    return `Welcome Player ${currentPlayer}. <br> You rolled ${player1Dice[0]} for the first 🎲 and ${player1Dice[1]} for the second 🎲.<br><br>Choose the order of the 🎲 by entering 1 or 2!`;
+  }
+  // wait for player 1 to input 1 or 2
+  // input validation
+  if (gameMode == "wait for player 1 choice") {
+    if (input != 1 && input != 2) {
+      return `Please choose either 1 or 2`;
+    }
+    gameMode = "player 2";
+    // call function and store player number according to input
+    // return message to let player know their number
+    player1Num = playerChoice(input);
+    console.log("player 1 number: " + player1Num);
+    return `Player ${currentPlayer}, you chose dice ${input} and your number is ${player1Num}<br> It is now player 2's turn`;
+  }
+  // start mode for player 2, generate roll dice for player 2
+  if (gameMode == "player 2") {
+    currentPlayer = 2;
+    newDiceRolls = getDiceRoll();
+    gameMode = "wait for player 2 choice";
+    return `Welcome Player ${currentPlayer}. <br> You rolled ${player2Dice[0]} for the first 🎲 and ${player2Dice[1]} for the second 🎲.<br><br>Choose the order of the 🎲 by entering 1 or 2!`;
+  }
+  // wait for player 2 to input 1 or 2
+  // input validation
+  if (gameMode == "wait for player 2 choice") {
+    if (input != 1 && input != 2) {
+      return `Please choose either 1 or 2`;
+    }
+    gameMode = "declare winner";
+    // call function and store player number according to input
+    // return message to let player know their number
+    player2Num = playerChoice(input);
+    console.log("player 2 number: " + player2Num);
+    return `Player ${currentPlayer}, you chose dice ${input} and your number is ${player2Num}<br> Click submit to see who is the winner`;
+  }
+  if (gameMode == "declare winner") {
+    // push player 1 and player 2 numbers into a new array
+    player1NumArray.push(Number(player1Num));
+    player2NumArray.push(Number(player2Num));
+    // sum up the numbers in the array after every round
+    // reset the sum of player numbers to 0
+    sumOfPlayer1Num = 0;
+
+    for (var i = 0; i < player1NumArray.length; i += 1) {
+      sumOfPlayer1Num += player1NumArray[i];
+    }
+    console.log("Player 1 total: " + sumOfPlayer1Num);
```

Comment:
it is good to use console log during development but remove them during code submission for clarity and readability of code